### ISSUE TITLE
Should `to_var_value`/`from_var_value` work with `std::vector`

### ISSUE THREAD

**Issue by bbbales2** [stan-dev/math#2297]
## Description

@SteveBronder, @t4c1, @rok-cesnovar 

We talk about these functions as if they convert any varmat-like input to matvar-like input and visa-versa, but they're only implemented for Stan vector/matrix/row vector types.

Should we include arrays of vectors/matrices/row vectors? It's not something we use at the Math level but it might be simpler to handle this in C++ than leaving it to stanc3 to generate conversion code.

A basic `to_var_value` for `std::vector` would look like:

```cpp
template <typename T>
auto to_var_value(const std::vector<T>& a) {
  std::vector<decltype(to_var_value(declval<T>()))> vec;
  for(size_t i = 0; i < a.size(); ++i) {
    vec.emplace_back(to_var_value(a[i]));
  }
  return vec;
}
```

This should probably forward it detects that the elements are already `var_value` types though.

And then the other direction would be similar.

Code for `to_var_value` [here](https://github.com/stan-dev/math/blob/develop/stan/math/rev/fun/to_var_value.hpp)
Code for `from_var_value` [here](https://github.com/stan-dev/math/blob/develop/stan/math/rev/fun/from_var_value.hpp)

#### Current Version:
v3.4.0

**Comment by rok-cesnovar** [comment]
Yes, I think we need this to make stanc3 code gen easier.

**Comment by bbbales2** [comment]
Closed by #2299
